fhir-persistence/src/main/java/com/ibm/fhir/persistence/index/FHIRRemoteIndexService.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * (C) Copyright IBM Corp. 2022
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+ 
+package com.ibm.fhir.persistence.index;
+
+import com.ibm.fhir.config.FHIRRequestContext;
+
+/**
+ * Service interface to support shipping resource search parameter values to an
+ * external service where they can be loaded into the database asynchronously.
+ * Implementations are expected to be tenant aware. They must use the tenantId
+ * value from the current {@link FHIRRequestContext}
+ */
+public abstract class FHIRRemoteIndexService {
+
+    // For now we just publish this as a static service
+    // TODO we should be injecting these services to something like the request context
```

Comment:
should we track this somewhere?

Reply:
I'll open an issue - it depends if we want to start using a different pattern overall for these types of services

Reply:
#3708